front/components/ConnectorPermissionsTree.tsx

```diff
@@ -122,6 +129,57 @@ function PermissionTreeChildren({
                   }
                 : undefined
             }
+            actions={
+              <div className="flex flex-row gap-2">
+                {r.lastUpdatedAt ? (
+                  <Tooltip
+                    contentChildren={
+                      <span className="mr-12">
+                        {new Date(r.lastUpdatedAt).toLocaleString()}
+                      </span>
+                    }
+                    position={i === 0 ? "below" : "above"}
+                  >
+                    <span className="text-xs text-gray-500">
+                      {timeAgoFrom(r.lastUpdatedAt)} ago
+                    </span>
+                  </Tooltip>
+                ) : null}
+                <IconButton
+                  size="xs"
+                  icon={ExternalLinkIcon}
+                  onClick={() => {
+                    if (r.sourceUrl) {
+                      window.open(r.sourceUrl, "_blank");
+                    }
+                  }}
+                  className={classNames(
+                    r.sourceUrl ? "" : "pointer-events-none opacity-0"
+                  )}
+                  disabled={!r.sourceUrl}
+                />
+                <IconButton
+                  size="xs"
+                  icon={BracesIcon}
+                  onClick={() => {
+                    if (r.dustDocumentId) {
+                      window.open(
```

Comment:
I worry this page might be too scary for an admin. Can we do a popup with a simple pre/code block in it?